### PR TITLE
Adding binding to see available deployments and fix binding return types

### PR DESCRIPTION
**Pull request by findolor**
> [!CAUTION]
> Do not merge before #1014

## Motivation

This PR allows users to view available deployments in the dotrain file and fixes the return types for some of the functions.

## Checks

By submitting this for review, I'm confirming I've done the following:
- [x] made this PR as small as possible
- [x] unit-tested any new functionality
- [ ] linked any relevant issues or PRs
- [ ] included screenshots (if this involves a front-end change)

### CODE REVIEW

File: crates/common/src/dotrain_order/mod.rs [hunk 1]
```diff
@@ -390,22 +397,55 @@ impl DotrainOrder {
...
+    fn update_config_source(
+        &mut self,
+        config_source: ConfigSource,
+    ) -> Result<(), DotrainOrderError> {
+        self.config_source = config_source.clone();
+        self.config = config_source.try_into()?;
+        Ok(())
+    }
+
+    pub fn update_config_source_bindings(
```
[the remark this answers is not in the document]
Comment (hardyjosh) [reply crates/common/src/dotrain_order/mod.rs:409]: look at the strategies on the prod webapp for examples

File: crates/common/src/dotrain_order/mod.rs [hunk 2]
```diff
@@ -427,13 +466,30 @@ impl DotrainOrder {
...
-            .insert(deployment_name.to_string(), Arc::new(deployment));
+        self.config_source
+            .orders
+            .insert(deployment.order.clone(), order.clone());
+        self.update_config_source(self.config_source.clone())?;
 
         Ok(())
     }
+
+    pub fn update_token_address(
```
Comment (hardyjosh) [inline crates/common/src/dotrain_order/mod.rs:477]: should this also update token infos afterwards?

File: crates/js_api/src/gui/mod.rs [hunk 3]
```diff
@@ -28,21 +29,44 @@ use wasm_bindgen::{
...
     deposits: Vec<deposits::TokenDeposit>,
+    select_tokens: Option<BTreeMap<String, Address>>,
     onchain_token_info: BTreeMap<Address, TokenInfo>,
 }
 #[wasm_bindgen]
 impl DotrainOrderGui {
-    #[wasm_bindgen(js_name = "init")]
-    pub async fn init(
+    #[wasm_bindgen(js_name = "getAvailableDeployments")]
+    pub async fn get_available_deployments(
```
Comment (hardyjosh) [inline crates/js_api/src/gui/mod.rs:58]: This should get GUI deployments

File: crates/common/src/dotrain_order/mod.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (findolor) [reply crates/common/src/dotrain_order/mod.rs:409]: fixed in #1014

File: crates/common/src/dotrain_order/mod.rs (hunk 2 above)
Comment (findolor) [reply crates/common/src/dotrain_order/mod.rs:477]: token information is updated in the gui library so it happens here https://github.com/rainlanguage/rain.orderbook/blob/44deaafa0d6b5d1ef921b95750e1725cff8ba045/crates/js_api/src/gui/select_tokens.rs#L53

**Review by hardyjosh — APPROVED** [review]

**Pull request merged**